tendrl/node_agent/manager/manager.py

```diff
@@ -331,15 +297,24 @@ def load_and_execute_sds_discovery_plugins(self):
 
 
 def main():
+    # Definitions
+    Tendrl.definitions = Tendrl.node_agent.objects.Definition()
+
+    # Register Config to tendrl.node_agent.objects namespace
+    Tendrl.config = Config()
+
     setup_logging(
-        config['log_cfg_path'],
-        config['log_level']
+        Tendrl.node_agent.objects.Config.data['log_cfg_path'],
+        Tendrl.node_agent.objects.Config.data['log_level']
     )
-# init definitions
-# init manager and object tree from etcd
-    machine_id = utils.get_machine_id()
 
-    m = NodeAgentManager(machine_id)
+    etcd_kwargs = {'port': config['etcd_port'],
+                   'host': config["etcd_connection"]}
+    Tendrl.etcd_orm = etcdobj.Server(etcd_kwargs=etcd_kwargs)
```

Comment:
One experience that I have had while working on monitoring related stuff is when the same etcd client is used for multiple operations to/from etcd, specially in case of blocking watch operation, if a subsquent read/write operation is to be made, using the same etcd_client object, both fail, so, I feel it might not be good to make global instance of etcd_orm because that might fail in case of concurrent etcd accesses.